### DISCUSSION TITLE
Basic authentication for Remotion Bundle/ServeUrl

### DISCUSSION THREAD

**Discussion by mnlbox** [remotion-dev/remotion#3710] [Q&A]
Hi @JonnyBurger ,

Is it possible to use Basic Authentication for a hosted [Remotion bundle](https://www.remotion.dev/docs/terminology/bundle) or [serveUrl](https://www.remotion.dev/docs/terminology/serve-url)?

**Comment by JonnyBurger** [comment]
I think it should work!
However, that is a server feature, so you need either:
-  nginx
- or on Netlify: https://docs.netlify.com/security/secure-access-to-sites/basic-authentication-with-custom-http-headers/
- or on Vercel: https://github.com/flawyte/vercel-basic-auth
- or on S3: https://stackoverflow.com/questions/3091084/does-amazon-s3-support-http-request-with-basic-authentication

Then just add the auth to the serve URL: `https://username:password@s3.amazonaws.com/...`

I never tried this though, but happy to make tweaks in case it doesn't work as expected.

**Comment by mnlbox** [comment]
Thanks @JonnyBurger

I checked the links that you shared.
I wanted to use Cloudflare Pages and as I checked implementing something like this is not simple without an extra Cloudflare worker.
I just wanted a very basic (and even less secure) password layer on top of my Remotion Bundle. After some research I think it's a bit over engineering for my project scope in the current stage.
Maybe I'll tackle it later and share my experience here.

Thanks for your support ❤️
